pennylane/transforms/qcut.py

```diff
@@ -1658,3 +1662,389 @@ def _infer_probed_cuts(
+    ``WireCut``s or ``MeasureNode``-``PrepareNode`` pairs.
+
+    Args:
+        graph (MultiDiGraph): The original (tape-converted) graph to be cut.
+
+    Returns:
+        (MultiDiGraph): Copy of the input graph with all its existing cuts removed.
+    """
+    uncut_graph = graph.copy()
+    for op in list(graph.nodes):
+        if isinstance(op, WireCut):
+            uncut_graph.remove_node(op)
+        elif isinstance(op, MeasureNode):
+            for op1 in graph.neighbors(op):
+                if isinstance(op1, PrepareNode):
+                    uncut_graph.remove_node(op)
+                    uncut_graph.remove_node(op1)
+
+    if len([n for n in graph.nodes if isinstance(n, (MeasureNode, PrepareNode))]) > 0:
+        warnings.warn(
+            "The circuit contains `MeasureNode`/`PrepareNode` that is not paired up correctly. "
+            "Please check.",
+            UserWarning,
+        )
+    return uncut_graph
+
+
+def find_and_place_cuts(
+    graph: MultiDiGraph, cut_method: Callable = kahypar_cut, replace_wire_cuts=False, **kwargs
+) -> MultiDiGraph:
+    """Automatically finds and places optimal ``WireCut`` nodes into a given tape-converted graph
+    using a customizable graph partitioning function. Preserves existing placed cuts.
+
+    Args:
+        graph (MultiDiGraph): The original (tape-converted) graph to be cut.
+        cut_method (Callable): A graph partitioning function that takes an input graph and returns
+            a list of edges to be cut based on a given set of constraints and objective. Defaults
+            to :func:`kahypar_cut` which requires KaHyPar to be installed: ``pip install kahypar``.
+        replace_wire_cuts (bool): Whether to replace ``WireCut`` nodes with ``MeasureNode`` and
+            ``PrepareNode`` pairs. Defaults to False.
+        kwargs: Additional keyword arguments to be passed to the callable ``cut_method``.
+
+    Returns:
+        (MultiDiGraph): Copy of the input graph with ``WireCut`` nodes inserted.
+
+    **Example**
+
+    Consider the following 4-wire circuit with a single CNOT gate connecting the top (wires
+    ``[1, 2]``) and bottom (wires ``["a", "b"]``) halves of the circuit. Note there's a ``WireCut``
+    manually placed into the circuit already.
+
+    .. code-block:: python
+
+        with qml.tape.QuantumTape() as tape:
+            qml.RX(0.1, wires=0)
+            qml.RY(0.2, wires=1)
+            qml.RX(0.3, wires="a")
+            qml.RY(0.4, wires="b")
+            qml.CNOT(wires=[0, 1])
+            qml.WireCut(wires=1)
+            qml.CNOT(wires=["a", "b"])
+            qml.CNOT(wires=[1, "a"])
+            qml.CNOT(wires=[0, 1])
+            qml.CNOT(wires=["a", "b"])
+            qml.RX(0.5, wires="a")
+            qml.RY(0.6, wires="b")
+            qml.expval(qml.PauliX(wires=[0]))
+            qml.expval(qml.PauliY(wires=["a"]) @ qml.PauliZ(wires=["b"]))
```

Comment:
Ah let me investigate 😂

Reply:
OK this is due to `_graph_to_hmetis` relying on the `order` attribute being unique, which can in fact be degenerate for tensor measurement nodes. Now the conversion is fixed by relying on the index of nodes instead of `order`. The random circuit test also added tensor expvals

Reply:
Thanks!